File: Backend-API/models/achievement.py
from datetime import datetime
import uuid

class Achievement:
    @staticmethod
    def create(name, description, points, icon):
        return {
            '_id': str(uuid.uuid4()),
            'name': name,
            'description': description,
            'points': points,
            'icon': icon,
            'created_at': datetime.now()
        }

File: RewardXLibrary/src/main/java/com/example/rewardxlibrary/models/Achievement.kt
package com.example.rewardxlibrary.models

import com.google.gson.annotations.SerializedName

data class Achievement(
    @SerializedName("_id")
    val id: String,

    @SerializedName("name")
    val name: String,

    @SerializedName("description")
    val description: String,

    @SerializedName("points")
    val points: Int,

    @SerializedName("icon")
    val icon: String,

    @SerializedName("created_at")
    val createdAt: String? = null
)